lib/ResolverFactory.js

```diff
@@ -218,9 +223,10 @@ exports.createResolver = function(options) {
 	modules.forEach(item => {
 		if (Array.isArray(item))
 			plugins.push(
-				new ModulesInHierachicDirectoriesPlugin("module", item, "resolve")
+				new ModulesInHierachicDirectoriesPlugin("module", item, "resolve", jail)
 			);
-		else plugins.push(new ModulesInRootPlugin("module", item, "resolve"));
+		else if (!jail || (jail && isSubDir(jail, item)))
+			plugins.push(new ModulesInRootPlugin("module", item, "resolve"));
```

Comment:
if item is not part of jail we should consider this a configuration error and throw here.

Reply:
The reason I had it like this is because I found that if one uses higher level projects like vue-cli or nuxt.js they set all sorts of paths in there and then this can just handled here transparently. But also happy to to fail as this can also be filtered out on the user side where we set the jail functionality.

Reply:
yep. Better fail to prevent questions like "Why does it ignore the value passed to the modules option?".